components/views.py

```diff
@@ -18,6 +20,10 @@ class ComponentListView(generics.ListCreateAPIView):
     permission_classes = [ComponentPermissions, ]
     filter_backends = [ComponentPermissionsFilter, ]
 
+    def filter_queryset(self, queryset: QuerySet) -> QuerySet:
+        # Need to apply the order_by after the union in ComponentPermissionsFilter
+        return super().filter_queryset(queryset).order_by("pk")
+
```

Comment:
Even though the tests passed, they could randomly fail because the `order_by` wasn't being preserved. Needs to be applied after the `union`, but I didn't think the `ComponentPermissionsFilter` applying an ordering that couldn't be changed was a good idea. Might need to think about this some more in the future, but I think this is fine for now.